tests/integration/__init__.py

```diff
@@ -389,6 +394,51 @@ def _id_and_mark(f):
 requires_custom_payload = pytest.mark.skipif(SCYLLA_VERSION is not None or PROTOCOL_VERSION < 4,
                                             reason='Scylla does not support custom payloads. Cassandra requires native protocol v4.0+')
 xfail_scylla = lambda reason, *args, **kwargs: pytest.mark.xfail(SCYLLA_VERSION is not None, reason=reason, *args, **kwargs)
+
+
+def xfail_scylla_version_lt(reason, oss_scylla_version, ent_scylla_version, *args, **kwargs):
+    """
+    It is used to mark tests that are going to fail on certain scylla versions.
+
+    ent_scylla_version should contain at least first major: `x.1.1` or `x.1` or `x`
+    """
+    if not reason.startswith("scylladb/scylladb#"):
+        raise ValueError('reason should start with scylladb/scylladb#<issue-id> to reference issue in scylla repo')
+
+    if not isinstance(ent_scylla_version, (tuple, list)):
+        ent_scylla_version = (ent_scylla_version,)
+    first_major = None
+    ent_scylla_version = [Version(v) for v in ent_scylla_version]
+
+    # Enterprise releases are tricky, here are the rules:
+    # 1. If something is fixed in `a.b.c` it is also fixed in `a.b.d` where c < d
+    # 2. If something is fixed in `a.1.1` it is also fixed in `c.d.f` where Version(a.b.1) < Version(`c.d.f`)
+    # To make version matching work properly of enterprise we need to have a first major version (`x.1.1`) where it was fixed, in addition to exact version.
+    for ent_v in ent_scylla_version:
+        if ent_v.minor in (1,0) and ent_v.micro in (1,0):
+            first_major = ent_v
```

Comment:
I'm not sure I get point 2. The variable `b` appear without conext and I don't know how to understand it and how to quantify it.
If we quantify it existentially (If something is fixed in `a.1.1` then it is also fixed in `c.d.f` if exists b such that Version(a.b.1) < Version(`c.d.f`)) it implies that if something is fixed in `2023.1.1` then it is also fixed in `2023.0.2` - because there exists such b (0) that `Version(2023.b.1) < Version(2023.0.2)`.

If we quantify it universally (If something is fixed in `a.1.1` then it is also fixed in `c.d.f` if for all b Version(a.b.1) < Version(`c.d.f`)) that means that if something is fixed in 2023.1.1 then it is not fixed in 2023.2.1. Why? Point 1 doesn't apply as minor versions don't match, and for b>=2 it is not true that `Version(2023.b.1) < Version(2023.2.1)`

Neither of those variants sounds probable.

Reply:
> If we quantify it universally (If something is fixed in `a.1.1` then it is also fixed in `c.d.f` if for all b Version(a.b.1) < Version(`c.d.f`)) that means that if something is fixed in 2023.1.1 then it is not fixed in 2023.2.1. Why? Point 1 doesn't apply as minor versions don't match, and for b>=2 it is not true that `Version(2023.b.1) < Version(2023.2.1)`

Mistake in comment, this one is correct.

Reason is this: `a.1.1` are LTS releases, which means that they are getting all major issues backported.
Now, say that something is fixed in `a.1.x`, where `x>1`, then it does not mean that it is fixed in any `a.2.z` where z >=1   

Let's take a look at two cases that I want to cover:
1. Issue is fixed in `a.1.1`, then it is safe to assume that it is  fixed in all versions older then `a.1.1`
2. Issue is fixed in `a.1.1`, and backported into `a-1.1.2`. Now you need a mechanism to let `pytest` know that it can run test for version `a-1.1.x` where `x>2` and on `b.c.d` where `b>=a` `c>=1` and `d>=1` .